src/PlatformEngines.jl

```diff
@@ -822,13 +844,28 @@ const CI_VARIABLES = [
     "TRAVIS",
 ]
 
-function get_telemetry_headers(url::AbstractString)
+const telemetry_file_lock = ReentrantLock()
+const telemetry_notice_printed = Ref(false)
+
+function get_telemetry_headers(url::AbstractString, notify::Bool=true)
     headers = String[]
-    server_dir = get_server_dir(url)
+    server = pkg_server()
+    server_dir = get_server_dir(url, server)
     server_dir === nothing && return headers
     push!(headers, "Julia-Pkg-Protocol: 1.0")
-    info = load_telemetry_file(joinpath(server_dir, "telemetry.toml"))
+    telemetry_file = joinpath(server_dir, "telemetry.toml")
+    notify &= !ispath(telemetry_file)
+    info = lock(telemetry_file_lock) do
+        load_telemetry_file(telemetry_file)
+    end
     get(info, "telemetry", true) == false && return headers
+    # legal (GDPR/CCPA) message about telemetry
+    if notify && !telemetry_notice_printed[]
+        telemetry_notice_printed[] = true
+        @info """
+        LEGAL NOTICE: package operations send anonymous data about your install to $server (your current package server), including the operating system and Julia versison you are running and a random client UUID. See https://julialang.org/legal/data/ for more details about what data is sent, what it is used for, how long it is retained, and how to opt out of sending this information.
```

Comment:
Worth referencing `Pkg.telemetryinfo()`?

Reply:
I was going to put that at the page that’s linked and go through each value that’s returned, what it means, why we collect it, and how to opt out of it. But maybe showing here would be helpful?

Reply:
Yea why not? It gives you a very quick access to the data that is sent and then you might not even bother visiting the website. `Pkg.telemetryinfo()` will also be more up-to-date compared to the website, and more accurate, since that is exactly what is sent whereas the website might not correspond to the same Julia version etc.

---

We should add what each value means to the output of `Pkg.telemetryinfo()` (or to its docstring).